src/bindings/swig/python/kdb.i

```diff
@@ -296,8 +303,18 @@
       return ret
 
     def extend(self, list):
+      """Extend the keyset by appending all the items from the iterable"""
       return self.append(*list)
 
+    def remove(self, name):
```

Comment:
Problems with rebase?

Reply:
I think you use `merge` instead of `rebase`, and some weird issues appear on github.

I keep my fork's master identical to upstream, and always rebase from master:
```
git checkout master
git fetch upstream master # assuming you have upstream set already
# do not do this if you ever work on your master branch
git reset --hard upstream/master # this is fine, I never work on master and I only want it to mirror upstream

git checkout mybranch
git rebase master
```
During this process you might have to fix some conflicts manually.
Maybe there are improvements to this process, but I think we **should** probably put this somewhere in the docs.

Reply:
I would also recommend, that the `master` branch of your fork is kept identical to that of the main repo.

I have set up my remotes like this (output of `git remote -vv`):
```
origin	git@github.com:kodebach/libelektra.git (fetch)
origin	git@github.com:kodebach/libelektra.git (push)
upstream	git@github.com:ElektraInitiative/libelektra (fetch)
upstream	git@github.com:ElektraInitiative/libelektra (push)
```

To update the `master` branch I do:

```sh
git fetch upstream master:master
git push origin master:master
```

This works without switching to the `master` branch first. After that you can do `git rebase master` to rebase the current branch.

Reply:
@kodebach can you copy what you wrote here to doc/GIT.md?

Reply:
Sure, but do we really need a git tutorial? Doesn't Elektra have enough barely maintained docs already?

Reply:
> @kodebach can you copy what you wrote here to doc/GIT.md?

See #3769